pyproject.toml

```diff
@@ -0,0 +1,43 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "qp_klp"
+# version strings must comply with PEP 440:
+# https://peps.python.org/pep-0440/
+version = "2022.4"
+authors = [
+    {name = "Qiita Development Team", email = "qiita.help@gmail.com"},
+]
+description = "Qiita Plugin: Knight Lab Processing"
+readme = "README.rst"
+# ">=" required as most available versions of 3.9 will be "3.9.17" or
+# similar.
+requires-python = ">=3.9"
+keywords = ["bioinformatics", "qiita", "sequence_processing_pipeline"]
+license = {text = "BSD"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: BSD License",
+    "Topic :: Scientific/Engineering :: Bio-Informatics",
+    "Topic :: Software Development :: Libraries :: Application Frameworks",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: MacOS :: MacOS X"
+    ]
+dependencies = [
+    "nose>=0.10.1",
+    "click>=3.3",
+    "future",
+    "pandas",
+    "qiita-files@https://github.com/qiita-spots/qiita-files/archive/master.zip",
+    "qiita_client@https://github.com/qiita-spots/qiita_client/archive/master.zip",
```

Comment:
Are these actually needed? I dont see them imported here or when I search the repo?

Reply:
We probably do need them, because we do have code that optionally connects to Qiita and compares set of sample-names against sample-names in Qiita and similar things. Note that this list of dependencies was taken directly from the setup.py that came before it.

Reply:
? https://github.com/biocore/mg-scripts/blob/master/setup.py#L36-L39

i didn't see any references to qiita client the code here too https://github.com/search?q=repo%3Abiocore%2Fmg-scripts%20client&type=code

Reply:
Agreed. These references can be seen in the pyproject.toml file found in qp-klp: https://github.com/charles-cowart/qp-knight-lab-processing/blob/80759dc1d776f8b99f87171a971e9fa3f3293e23/pyproject.toml#L32-L39. They were not changed/removed when using this file as the template for this project's pyproject.toml file.

Reply:
For the moment this file has been removed.